__tests__/p2sh/tx-proposal-melt-tokens.test.js

```diff
@@ -0,0 +1,249 @@
+    await TestUtils.stopWallet({ walletId });
+  });
+
+  it('should return 200 with a valid body', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 1,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.txHex).toBeDefined();
+    const tx = hathorLib.helpersUtils
+      .createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+    expect(tx.outputs.map(o => o.decodedScript.address.base58))
+      .toEqual(expect.arrayContaining(['wbe2eJdyZVimA7nJjmBQnKYJSXmpnpMKgG']));
+  });
+
+  it('should not accept melt token with empty token', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '',
+        amount: 1,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should not accept melt token with amount 0', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 0,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should not accept melt token without funds to cover it', async () => {
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 1_000_000_000_000, // 1 trillion
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should return 200 with a valid body selecting deposit address', async () => {
+    const depositAddress = TestUtils.multisigAddresses[2];
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/melt-tokens')
+      .send({
+        token: '0000073b972162f70061f61cf0082b7a47263cc1659a05976aca5cd01b3351ee',
+        amount: 100,
+        deposit_address: depositAddress,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.txHex).toBeDefined();
+    const tx = hathorLib.helpersUtils.createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+    expect(tx.outputs.map(o => o.decodedScript.address.base58))
+      .toEqual(expect.arrayContaining(['wbe2eJdyZVimA7nJjmBQnKYJSXmpnpMKgG', depositAddress]));
+    expect(tx.inputs).toEqual(expect.not.arrayContaining([
```

Comment:
This is coming with the input data so the tests are failing, is this fixed on another PR?

Reply:
Yes, it will be fixed when we release the next wallet-lib version.